electron/src/main.common.ts

```diff
@@ -1,21 +1,65 @@
-const defaultUserDataPath   = app.getPath("home") + "/.sevenbridges/rabix-composer";
-const webdriverUserDataPath = defaultUserDataPath + "-e2e";
+const isSpectronRun       = ~process.argv.indexOf("--spectron");
+const defaultUserDataPath = app.getPath("home") + "/.sevenbridges/rabix-composer";
+
+app.setPath("userData", defaultUserDataPath);
+
+/**
+ * If we are running functional tests with spectron, we need to expose more of the app to
+ * outside control.
+ *
+ * First, we need to be able to override the directory in Chromium will be storing data.
+ * This is “--user-data-dir”. We need it so tests will not store data in the same folder where the
+ * regular apps stores it, and tests should not share data amongst themselves, so each test case
+ * will provide a different directory, and clean it up afterwards.
+ *
+ * We also need a possibility to override some modules in the app.
+ * For example, we don't want to do actual HTTP requests. We instead need a way to mock
+ * modules on per-test basis.
+ *
+ */
+if (isSpectronRun) {
+    // Define argument names that we are looking for
+    const dirArgName             = "--user-data-dir=";
+    const moduleOverridesArgName = "--override-modules=";
+
+    // Find if arguments are present in the command line
+    const userDataDirArg     = process.argv.find(arg => arg.startsWith(dirArgName));
+    const moduleOverridesArg = process.argv.find(arg => arg.startsWith(moduleOverridesArgName))
+
+    // If we're given an alternate userData directory, override the default one
+    if (userDataDirArg) {
+        const userDir = userDataDirArg.slice(dirArgName.length);
+        app.setPath("userData", userDir);
+    }
+
+    // If we're given module overrides, we're given a string through the command line
+    // so we need to unpack it
+    if (moduleOverridesArg) {
+        // Take the argument value
+        const serializedOverrides = moduleOverridesArg.slice(moduleOverridesArgName.length);
+
+        // Deserialize it in such way that everything that is not an object goes through eval
+        // We serialized function as strings beforehand, so we need to bring them back to life
+        const overrides = JSON.parse(serializedOverrides, (key, val) => {
+            if (typeof val !== "object") {
+                return eval(`(${val})`);
+            }
 
+            return val;
+        });
 
-if (isWebdriverRun) {
-    mkdirp.sync(webdriverUserDataPath);
-    app.setPath("userData", webdriverUserDataPath);
-    global["__endpoints"]        = require("./routes");
-    global["__webdriverCleanup"] = () => rimraf(webdriverUserDataPath, () => void 0);
-} else {
-    app.setPath("userData", defaultUserDataPath);
+        // For all modules that should be mocked, provide given mocks to the module loader now,
+        // before anybody else requires them.
+        // That way, they will get mocks from cache.
+        for (let moduleName in overrides) {
```

Comment:
If I understood correctly, this is the `overrideModules` argument in the boot function. Seems to me the keys are more `modulePath`s than `moduleName`s